[Scrabble Score]: approaches (#3320)
exercism/python: exercises/practice/scrabble-score/.approaches/two-sequences/content.md

# Two sequences

```python
KEYS = "AEIOULNRSTDGBCMPFHVWYKJXQZ"
SCORES = [1] * 10 + [2] * 2 + [3] * 4 + [4] * 5 + [5] * 1 + [8] * 2 + [10] * 2

> def score(word):
>     return sum(SCORES[KEYS.index(letter.upper())] for letter in word)
> ```

Reviewer: ```suggestion
def score(word):
    return sum(SCORES[KEYS.index(letter)] for letter in word.upper())
```
```

> This approach uses a string and a [list][list], both of these data types belongs to the parent data type [sequences][sequence].

Reviewer: ```suggestion
This approach uses a string and a [list][list], both of which are [sequence][sequence] types.
```

> The code starts with defining a string constant with letters.

Reviewer: ```suggestion
The code begins by defining a string constant with letters.
```

> Then another constant is definded which is a list with corresponding score for the same index as the string.

Reviewer: ```suggestion
Then another constant is defined as a list with the corresponding letter score at the same index as the letter in the string.
```

> The `score` function takes a word as a parameter.

Reviewer: ```suggestion
The `score` function takes a word as an argument.
```

> And uses the same [generator expression][generator-expersion] as the [dictionary approach][dictionary-approach] with some slight modifications.

Reviewer: ```suggestion
And uses the same [generator expression][generator-expression] as the [dictionary approach][dictionary-approach] with some slight modifications.
```

> The difference is that instead of using a [dictionary][dictionary] and looking up the score inside of a dictonary.
> This approach gets the index of the letter in the KEYS constant and then then looks up the value for that index in SCORES list.
> Then takes that value and return that to the generator expression.

Reviewer: ```suggestion
Instead of using a [dictionary][dictionary] and looking up the score, this approach looks up the index of the letter in the KEYS constant and then then looks up the value for that index in SCORES list within the generator expression.
These values are then added up by `sum`.
```
